ntbs-service/Jobs/ReportingDataRefreshJob.cs

```diff
@@ -1,123 +1,51 @@
-        : base(configuration)
+        public ReportingDataRefreshJob(IExternalStoredProcedureRepository externalStoredProcedureRepository)
         {
-            _reportingDatabaseConnectionString = _configuration.GetConnectionString(Constants.DbConnectionStringReporting);
-            _specimenMatchingConnectionString = _configuration.GetConnectionString(Constants.DbConnectionStringSpecimenMatching);
-            _migrationConnectionString = _configuration.GetConnectionString(Constants.DbConnectionStringMigration);
-
-            _sqlString = "";
-            _parameters = null;
+            _externalStoredProcedureRepository = externalStoredProcedureRepository;
         }
 
-        /// PerformContext context is passed in via Hangfire Server
-        public override async Task Run(PerformContext context, IJobCancellationToken token)
+        // PerformContext context is passed in via Hangfire Server
+        public async Task Run(PerformContext context)
         {
-            _context = context;
-            Log.Information($"Starting reporting data refresh job.");
+            LogInfo(context, "Starting reporting data refresh job");
 
             try
             {
-                var stepOneResults = await ExecuteSpecimenMatchingGenerateStoredProcedure(token);
-                var stepTwoResults = await ExecuteReportingGenerateStoredProcedure(token);
-                var stepThreeResults = await ExecuteMigrationGenerateStoredProcedure(token);
-
-                var allResults = new List<dynamic>();
-                allResults.AddRange(stepOneResults);
-                allResults.AddRange(stepTwoResults);
-                allResults.AddRange(stepThreeResults);
-
-                var success = DidExecuteSuccessfully(allResults);
+                LogInfo(context, "Starting specimen-matching uspGenerate");
+                var stepOneResults =
+                    await _externalStoredProcedureRepository.ExecuteSpecimenMatchingGenerateStoredProcedure();
+                CheckExecutedSuccessfully(context, stepOneResults);
+
+                LogInfo(context, "Starting reporting uspGenerate");
+                var stepTwoResults =
+                    await _externalStoredProcedureRepository.ExecuteReportingGenerateStoredProcedure();
+                CheckExecutedSuccessfully(context, stepTwoResults);
+
+                LogInfo(context, "Starting migration uspGenerate");
+                var stepThreeResults =
+                    await _externalStoredProcedureRepository.ExecuteMigrationGenerateStoredProcedure();
+                CheckExecutedSuccessfully(context, stepThreeResults);
             }
             catch (Exception ex)
             {
-                _context.WriteLine(ex.Message);
-                Log.Error(ex, "Error occured during reporting data refresh job.");
+                context.WriteLine(ex.Message);
+                Log.Error(ex, "Error occured during reporting data refresh job");
                 throw;
```

Comment:
I think the fact that we log the error and then rethrow it means it will end up being logged twice, which isn't ideal. However, since it's a pre-existing pattern and the worst case scenario is over-logging, I'm happy to investigate that at a later date.